pages/k8s/monitoring.md

```diff
@@ -0,0 +1,257 @@
+---
+
+# Monitoring
+
+The **Canonical Distribution of Kubernetes**<sup>&reg;</sup> includes the
+standard **Kubernetes** dashboard for monitoring your cluster. However, it is
+often advisable to have a monitoring solution which will run whether the
+cluster itself is running or not. It may also be useful to integrate monitoring
+into existing setups.
+
+**Prometheus** is the recommended way to monitor your deployment -
+instructions are provided below. There are also instructions for setting up
+other monitoring solutions, or connecting to existing monitoring setups.
+
+## Monitoring with Prometheus
+
+The recommended way to monitor your cluster is to use a combination of
+**Prometheus**, **Grafana** and **Telegraf**. The fastest, easiest way to
+install and configure this is via **conjure-up** when installing the
+**Canonical Distribution of Kubernetes**<sup>&reg;</sup>, by selecting the box
+next to  Prometheus from the `Add-on` menu. You can then log in to the dashboard
+as [described below](#retrieve-credentials-and-login). See the [quickstart
+guide][quickstart] for more details on installing **CDK** with **conjure-up**.
+
+If you have already installed your cluster, you will be able to add and
+configure  the extra applications using **Juju** as described here:
+
+### Install the required applications
+
+The following commands will add the required applications:
+
+```bash
+juju deploy grafana --series=xenial
+juju deploy prometheus  --series=xenial
+juju deploy telegraf  --series=xenial
+juju expose grafana
+```
+
+### Add relations
+
+Relationships need to be established between the applications and the units
+running the cluster:
+
+```bash
+juju add-relation prometheus:grafana-source grafana:grafana-source
+juju add-relation telegraf:prometheus-client prometheus:target
+juju add-relation kubernetes-master:juju-info telegraf:juju-info
+juju add-relation kubernetes-worker:juju-info telegraf:juju-info
+```
+
+### Adding a scraper for Prometheus
+
+Prometheus will also need an appropriate scraper to collect metrics relevant to
+the cluster.  A useful default is installed when using **conjure-up** (the
+template for this can be [downloaded here][download-scraper]), but you can also
+configure it manually by following the steps outlined here:
+
+ -  Download the scraper file
+    ```bash
+    curl -O  https://raw.githubusercontent.com/conjure-up/spells/master/canonical-kubernetes/addons/prometheus/steps/01_install-prometheus/prometheus-scrape-k8s.yaml
+    ```
+    This is the template, but it needs some specific information for your cluster.
+ -  Get the relevant address and password form your cluster
+     ```bash
+    api=$(juju run  --unit kubeapi-load-balancer/0 'network-get website --format yaml --ingress-address' | head -1)
+    pass=$(grep 'password:' ~/.kube/config | sed -e 's/ *//g' -e 's/password://')
```

Comment:
This may return multiple passwords if the user has a lot of k8s clusters.  Let's get the true value straight out of the master, and use awk instead of those wacky seds:
```diff
- pass=$(grep 'password:' ~/.kube/config | sed -e 's/ *//g' -e 's/password://')
+ pass=$(juju run --unit kubernetes-master/0 'grep "password:" /home/ubuntu/config' | awk '{ print $2 }')
```